vxtelegram/telegram.py

```diff
@@ -157,31 +185,104 @@ def handle_outbound_message(self, message):
         url = self.get_outbound_url('sendMessage')
         http_client = HTTPClient(self.agent_factory())
 
-        try:
-            r = yield http_client.post(
-                url=url,
-                data=json.dumps(outbound_msg),
-                headers={'Content-Type': ['application/json']},
-            )
+        r = yield http_client.post(
+            url=url,
+            data=json.dumps(outbound_msg),
+            headers={'Content-Type': ['application/json']},
+            allow_redirects=False,
+        )
+
         # Telegram redirects our request if our bot token is invalid
-        except ResponseFailed as e:
-            yield self.outbound_failure(message_id,
-                                        'Invalid token (redirected)\n%s' % e)
+        if r.code == http.FOUND:
+            yield self.outbound_failure(
+                message_id=message_id,
+                message='Invalid bot token (redirected)',
+                error='Redirected',
+            )
             return
+
         try:
             res = yield r.json()
-        except ValueError as e:
-            yield self.outbound_failure(message_id,
-                                        'Expected JSON response\n%s' % e)
+        except ValueError:
+            yield self.outbound_failure(
+                message_id=message_id,
+                message='Expected JSON response',
+                error='error code %s' % r.code,
+            )
             return
 
         if r.code == http.OK and res['ok']:
-            yield self.publish_ack(user_message_id=message_id,
-                                   sent_message_id=message_id)
+            yield self.outbound_success(message_id)
         else:
-            yield self.outbound_failure(message_id, res['description'])
+            yield self.outbound_failure(
+                message_id=message_id,
+                message='Bad response from Telegram',
+                error=res['description'],
```

Comment:
I think it would be helpful to include the (telegram error) code and description as `details` fields in the status message.